itests/fevm_test.go

```diff
@@ -1659,3 +1660,100 @@ func TestEthEstimateGas(t *testing.T) {
 		})
 	}
 }
+
+func TestEthNullRoundHandling(t *testing.T) {
+	blockTime := 100 * time.Millisecond
+	client, _, ens := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC())
+
+	bms := ens.InterconnectAll().BeginMining(blockTime)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	bms[0].InjectNulls(10)
+
+	tctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	ch, err := client.ChainNotify(tctx)
+	require.NoError(t, err)
+	<-ch
+	hc := <-ch
+	require.Equal(t, store.HCApply, hc[0].Type)
+
+	afterNullHeight := hc[0].Val.Height()
+
+	nullHeight := afterNullHeight - 1
+	for nullHeight > 0 {
+		ts, err := client.ChainGetTipSetByHeight(ctx, nullHeight, types.EmptyTSK)
+		require.NoError(t, err)
+		if ts.Height() == nullHeight {
+			nullHeight--
+		} else {
+			break
+		}
+	}
+
+	nullBlockHex := fmt.Sprintf("0x%x", nullHeight)
```

Comment:
```suggestion
	nullBlockHex := fmt.Sprintf("0x%x", int(nullHeight))
```

this is the real bug you're encountering - `%x` is doing a string version of the argument, but it's an `abi.ChainEpoch` which has its own `String()` method. So we'll just make sure `Sprintf` knows it's just an `int`. If you print your output you'll see it's not what you want and instead it's encoding the string version of the epoch as the bytes of that string, which becomes a much bigger hex number than you want it to be.